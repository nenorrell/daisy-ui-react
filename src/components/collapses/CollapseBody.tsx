import clsx from "clsx";
import { ForwardedRef, forwardRef, PropsWithChildren } from "react";

export interface ICollapseBody {
    isExpanded ?:boolean,
    className ?:string
}
export const CollapseBody = forwardRef((
    props :PropsWithChildren<ICollapseBody>,
    ref ?:ForwardedRef<HTMLDivElement>
) => {
    return (
        <div
            ref={ref}
            className={clsx(
                "collapse-body",
                "ease-linear",
                "duration-[.25s]",
                "transition-[height]",
                "overflow-hidden",
                !props.isExpanded && "h-0",
                props.className
            )}
        >
            {props.children}
        </div>
    );
});

CollapseBody.displayName = "CollapseBody";
