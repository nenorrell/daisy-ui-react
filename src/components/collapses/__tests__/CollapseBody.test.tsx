import { render } from "@testing-library/react";
import { CollapseBody } from "../CollapseBody";

describe("SimpleCollapse", ()=>{
    it("Should render multiple children as expected", ()=>{
        const {container} = render(
            <CollapseBody>
                <h2 className="w-full text-2xl">This is the body</h2>
                <div className="mr-4">something else</div>
            </CollapseBody>
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render single child as expected", ()=>{
        const {container} = render(
            <CollapseBody>
                <h2 className="w-full text-2xl">This is the body</h2>
            </CollapseBody>
        );
        expect(container).toMatchSnapshot();
    });

    it("Respects the className prop", ()=>{
        const {container} = render(
            <CollapseBody className="find-me">
                <h2 className="w-full text-2xl">This is the body</h2>
            </CollapseBody>
        );
        expect(container.querySelector("div.find-me")).toBeInTheDocument();
    });

    it("Does not shrink body when expanded", ()=>{
        const {container} = render(
            <CollapseBody isExpanded>
                <h2 className="w-full text-2xl">This is the body</h2>
            </CollapseBody>
        );
        expect(container.querySelector("div.collapse-body")).not.toHaveClass("h-0");
    });

    it("Does shrink body when not expanded", ()=>{
        const {container} = render(
            <CollapseBody>
                <h2 className="w-full text-2xl">This is the body</h2>
            </CollapseBody>
        );
        expect(container.querySelector("div.collapse-body")).toHaveClass("h-0");
    });
});
