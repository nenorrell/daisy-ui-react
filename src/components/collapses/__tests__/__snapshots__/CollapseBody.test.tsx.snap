// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCollapse Should render multiple children as expected 1`] = `
<div>
  <div
    class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0"
  >
    <h2
      class="w-full text-2xl"
    >
      This is the body
    </h2>
    <div
      class="mr-4"
    >
      something else
    </div>
  </div>
</div>
`;

exports[`SimpleCollapse Should render single child as expected 1`] = `
<div>
  <div
    class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0"
  >
    <h2
      class="w-full text-2xl"
    >
      This is the body
    </h2>
  </div>
</div>
`;
